src/he/msg_handlers.c

```diff
@@ -387,6 +388,15 @@ static he_return_code_t internal_handle_data(he_conn_t *conn, uint8_t *inside_pa
     return HE_ERR_BAD_PACKET;
   }
 
+  // Clamp the MSS if PMTU has been fixed
+  if(conn->pmtud_state == HE_PMTUD_STATE_SEARCH_COMPLETE) {
+    uint16_t effective_pmtu = he_conn_get_effective_pmtu(conn);
+    he_return_code_t ret = he_clamp_mss(inside_packet, pkt_length, effective_pmtu - HE_MSS_OVERHEAD);
+    if(ret != HE_SUCCESS) {
+      return ret;
+    }
+  }
```

Comment:
Probably no need to do mss clamping again here. Because:

* If it's server code, the packet is already "mss clamped" by client in above he_conn_inside_packet_received
* If it's client code, this code should never be TCP SYN packe (because all the tcp connections are initiated by the client), so there's no need to do mss clamping

Reply:
So are we only going to clamp syn packets client side for now? Since we do not have server to client discovery yet

Reply:
We discussed this offline and it has been decided we will implement server side clamping as well.